### PR TITLE
POST | Cart

### PR DESCRIPTION
**Pull request by Mu7ammadAbed**
- I handled `/cart` route with `POST` method which add products into customer cart.

- I made a function for `/cart` route.

- I made a database query which `SELECT` all rows from cart table `WHERE` their values match the customer id and product id.

- If there's any data stored in the database with the entered id's, the controller function will directly throw an error to user that this product is in your cart already.

- If the product isn't in customer cart, then it will `INSERT` it's enterd values into database.

- I removed `price` from `cart` table because it's not necessary, we can access product price at any time using `painting_id` which is already exist in `cart` table and because product price may change by time at `painting` table while it's still the same as before at `cart` table which will cause many problems.

- I tested all the cases (Success and Failing).

- I made a validation schema for `customer_id` and `painting_id`.

### CODE REVIEW

File: server/utils/validation/cartSchema.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+const yup = require('yup');
```
Comment (mariamisa) [inline server/utils/validation/cartSchema.js:1]: destruct yup 
const {object , number} = require('yup');

File: server/controllers/index.js [hunk 2]
```diff
@@ -1,6 +1,7 @@
 const error = require('./errorHandlers');
-const getPaintings = require('./getPaintings');
+const { getPaintings } = require('./Paintings');
+const { postCart } = require('./Cart');
```
Comment (mariamisa) [inline server/controllers/index.js:3]: change name to (addPaintingToCart )

File: server/controllers/Cart/postCart.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  try {
+    await cartSchema.validate(req.body, { abortEarly: false });
+    const { rows: checkrows } = await checkCartQuery(req.body);
+    if (checkrows.length === 0) {
+      const { rows } = await postCartQuery(req.body);
+      const { customer_id: customerId, painting_id: paintingId } = rows[0];
+      res.status(201).json({
+        StatusCode: 201,
+        data: {
+          customerId,
```
Comment (LinaYahya) [inline server/controllers/Cart/postCart.js:14]: customerId it's not at the request body, it exists at the token.

File: server/routes/cart.js [hunk 4]
```diff
@@ -0,0 +1,7 @@
+const router = require('express').Router();
+
+const { postCart } = require('../controllers');
+
+router.post('/cart', postCart);
```
Comment (LinaYahya) [inline server/routes/cart.js:5]: This is a protected route, so verify user first by middleware.

**Pull request merged**